### PR TITLE
e2e: Add `cluster-info` as job artifact

### PR DESCRIPTION
**Pull request by zeeke**
Use `kubectl cluster-info dump` to retrieve useful debug information at the end of the end-to-end test run.

### CODE REVIEW

**Review by SchSeba — APPROVED** [review]
Thanks!

/lgtm

File: hack/run-e2e-conformance-virtual-cluster.sh
Comment (mlguerrero12) [inline hack/run-e2e-conformance-virtual-cluster.sh:369]: is `pwd` not longer needed?

Comment (zeeke) [reply hack/run-e2e-conformance-virtual-cluster.sh]: Good catch: relative paths are relatives to the test suite entrypoint, i.e. `test/conformance/test_suite_test.go`

**Comment by mlguerrero12** [comment]
LGTM

**Comment by zeeke** [comment]
/hold

artifact archives does not contains junit XML report and failure folder (there should be at least an empty folder)
https://github.com/k8snetworkplumbingwg/sriov-network-operator/actions/runs/6704825113?pr=534

**Review by adrianchiris — APPROVED** [review]

**Comment by zeeke** [comment]
Got 3 LGTM from 2 different vendors, merging

**Pull request merged**
